launchdarkly-server-sdk/src/migrations/migrator.rs

```diff
@@ -43,79 +43,63 @@ pub struct MigrationWriteResult<T> {
 // provided results are equal, this method will return true and false otherwise.
 type MigrationComparisonFn<T> = fn(&T, &T) -> bool;
 
-struct MigrationConfig<T, FO, FN>
+struct MigrationConfig<P, T, FO, FN>
```

Comment:
Originally this was requiring the `read` or `write` `payload` parameter to be the same type as the return of those methods. That isn't a requirement of the spec so we should reflect that appropriately. So lots of type changes here.